text/3660-crates-io-crate-deletions.md

```diff
@@ -0,0 +1,123 @@
+- Start Date: 2024-06-20
+- RFC PR: [rust-lang/rfcs#3660](https://github.com/rust-lang/rfcs/pull/3660)
+
+# Summary
+
+This RFC proposes a mechanism for crate authors to delete their crates from crates.io under certain conditions.
+
+
+# Motivation
+
+There are a variety of reasons why a crate author might want to delete a crate or version from crates.io:
+
+* You published something accidentally.
+* You wanted to test crates.io.
+* You published content you didn't intend to be public.
+* You want to rename a crate. (The only way to rename a package is to re-publish it under a new name)
+
+The current [crates.io usage policy](https://crates.io/policies) says:
+
+> Crate deletion by their owners is not possible to keep the registry as immutable as possible.
+
+This restriction makes sense for the majority of crates that have been around for a while and are actively used, but the above list of reasons shows that there are valid use cases for allowing crate authors to delete their crates without having to contact the crates.io team.
+
+To make this process easier for our users and to reduce the workload of the crates.io team dealing with such support requests, we propose to codify our current set of informal rules into a formal policy that allows crate authors to delete their crates themselves under certain conditions (see below).
+
+
+# Proposal
+
+We propose to allow crate authors to delete their **crates** from crates.io under the following conditions:
+
+* The crate has been published for less than 72 hours,
+* or if all the following conditions are met:
+  * The crate has a single owner,
+  * The crate has no reverse dependencies,
+  * The crate has been downloaded less than 100 times for each month it has been published.
```

Comment:
Maybe limit the time frame to "in the last year" or so? Crates may have had more users before they were renamed/obsoleted. 

100 is too low - that's bot traffic background noise level.

Reply:
> Maybe limit the time frame to "in the last year" or so? Crates may have had more users before they were renamed/obsoleted.

that would make it quite hard to go back to a previous revision of your projects from years ago that used to use that dependency. I think I'd prefer crates that had been used quite a bit a while ago to be forced to stay around.

> 100 is too low - that's bot traffic background noise level.

the spam/name-squatting/placeholder crates I've removed over the past months usually had a couple of hundred downloads in total and were usually a year old or even more. admittedly, our download counting method changed lately which increased the numbers. as mentioned in the RFC text, we can always adjust the numbers later if we see that they are indeed to low (or high).